### PR TITLE
Aof rewrite fix and test

### PR DESCRIPTION
**Pull request by alonre24**
Refactoring and testing Redis AOF_Rewrite mechanism on ReidsAI types (probably for the first time...)

### CODE REVIEW

File: src/serialization/AOF/rai_aof_rewrite.c [hunk 1]
```diff
@@ -75,29 +62,54 @@ void RAI_AOFRewriteModel(RedisModuleIO *aof, RedisModuleString *key, void *value
...
-    RedisModule_EmitAOF(aof, "AI.MODELSET", "slccclclcvcvcv", key, backendstr, model->devicestr,
-                        model->tag, "BATCHSIZE", model->opts.batchsize, "MINBATCHSIZE",
-                        model->opts.minbatchsize, "INPUTS", inputs_, model->ninputs, "OUTPUTS",
-                        outputs_, model->noutputs, "BLOB", buffers_, n_chunks);
+    if (model->backend != RAI_BACKEND_TENSORFLOW) {
+
+        RedisModule_EmitAOF(aof, "AI.MODELSTORE", "scccsclclclcv", key, backendstr,
+                            model->devicestr, "TAG", model->tag, "BATCHSIZE", model->opts.batchsize,
+                            "MINBATCHSIZE", model->opts.minbatchsize, "MINBATCHTIMEOUT",
+                            model->opts.minbatchtimeout, "BLOB", buffers_, n_chunks);
```
Comment (DvirDukhan) [inline src/serialization/AOF/rai_aof_rewrite.c:70]: why do we need n_chunks?

File: src/serialization/AOF/rai_aof_rewrite.c [hunk 2]
```diff
@@ -75,29 +62,54 @@ void RAI_AOFRewriteModel(RedisModuleIO *aof, RedisModuleString *key, void *value
...
 
-    RedisModule_EmitAOF(aof, "AI.MODELSET", "slccclclcvcvcv", key, backendstr, model->devicestr,
-                        model->tag, "BATCHSIZE", model->opts.batchsize, "MINBATCHSIZE",
-                        model->opts.minbatchsize, "INPUTS", inputs_, model->ninputs, "OUTPUTS",
-                        outputs_, model->noutputs, "BLOB", buffers_, n_chunks);
+    if (model->backend != RAI_BACKEND_TENSORFLOW) {
+
+        RedisModule_EmitAOF(aof, "AI.MODELSTORE", "scccsclclclcv", key, backendstr,
+                            model->devicestr, "TAG", model->tag, "BATCHSIZE", model->opts.batchsize,
+                            "MINBATCHSIZE", model->opts.minbatchsize, "MINBATCHTIMEOUT",
```
Comment (DvirDukhan) [inline src/serialization/AOF/rai_aof_rewrite.c:69]: are we guaranteed that those values have defaults?

File: src/serialization/AOF/rai_aof_rewrite.c [hunk 3]
```diff
@@ -75,29 +62,54 @@ void RAI_AOFRewriteModel(RedisModuleIO *aof, RedisModuleString *key, void *value
...
+                            model->opts.minbatchtimeout, "BLOB", buffers_, n_chunks);
+    } else {
+        // For TF backend, the command should contain INPUTS and OUTPUTS names.
+        // Create RedisModuleString* arrays from the char* arrays, so we can send a proper vector
+        // to RedisModule_EmitAOF.
+        RedisModuleString **inputs_ = array_new(RedisModuleString *, model->ninputs);
+        RedisModuleString **outputs_ = array_new(RedisModuleString *, model->noutputs);
```
Comment (DvirDukhan) [inline src/serialization/AOF/rai_aof_rewrite.c:76]: new on the stack?

File: tests/flow/test_serializations.py [hunk 4]
```diff
@@ -216,3 +217,74 @@ def test_v2_tensor(self):
         self.env.assertEqual([tensor_type, tensor_shape], [b"INT32", [2, 1]])
         values = con.execute_command('AI.TENSORGET', key_name, 'VALUES')
         self.env.assertEqual(values, [1, 2])
+
+
+class TestAofRewrite:
```
Comment (DvirDukhan) [inline tests/flow/test_serializations.py:222]: please try testing with different configuration model configuration params absent (that yields legal commands)

**Review by DvirDukhan — CHANGES_REQUESTED** [review]

File: src/serialization/AOF/rai_aof_rewrite.c (hunk 1 above)
Comment (alonre24) [reply src/serialization/AOF/rai_aof_rewrite.c:70]: in the `fmt` arg the last char is `v`, which means that we should pass a vector (array) of `RM_String` followed by its size.

File: src/serialization/AOF/rai_aof_rewrite.c (hunk 2 above)
Comment (alonre24) [reply src/serialization/AOF/rai_aof_rewrite.c:69]: yes

File: src/serialization/AOF/rai_aof_rewrite.c (hunk 3 above)
Comment (alonre24) [reply src/serialization/AOF/rai_aof_rewrite.c:76]: I'll change it (it will require using an arbitrary const as `cap`)

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**